File: pmoqobuz/test_python_qobuz/test_getfileurl.py
#!/usr/bin/env python3
"""
Test simplifié - va directement à track_getFileUrl
Pour utiliser avec le fake server
"""

import sys
import getpass
from raw import RawApi

def main():
    print("=== Test track_getFileUrl (Python) ===\n")

    # Credentials
    username = input("Qobuz username (default: eric@coissac.eu): ").strip() or "eric@coissac.eu"
    password = getpass.getpass("Qobuz password: ")

    if not password:
        print("Error: password required")
        return 1

    # Track ID connu (récupéré du test précédent)
    track_id = "19557883"
    format_id = 27

    print("1. Creating RawApi (auto-initializes Spoofer)...")
    api = RawApi(appid=None, configvalue=None)
    print(f"   App ID: {api.appid}")

    print("\n2. Logging in...")
    login_result = api.user_login(username=username, password=password)
    if not login_result:
        print(f"   ✗ Login failed: {api.error}")
        return 1
    print(f"   ✓ Login successful - User ID: {api.user_id}")
    print(f"   Token: {api.user_auth_token[:20]}...")

    # Appel direct à track_getFileUrl
    print(f"\n3. Calling track_getFileUrl...")
    print(f"   Track ID: {track_id}")
    print(f"   Format ID: {format_id}")
    print(f"   ⚠️  THIS CALL IS SIGNED - Watch fake server logs!\n")

    file_url_data = api.track_getFileUrl(track_id=track_id, format_id=format_id)

    if not file_url_data:
        print(f"   ✗ Failed: {api.error}")
        print(f"   Status code: {api.status_code}")
        return 1

    url = file_url_data.get('url', '')
    mime_type = file_url_data.get('mime_type', '')

    print(f"   ✓ Success!")
    print(f"   URL: {url[:80]}...")
    print(f"   MIME type: {mime_type}")

    print("\n=== Test completed successfully! ===")
    return 0

if __name__ == "__main__":
    sys.exit(main())

File: pmoqobuz/examples/test_getfileurl.rs
//! Test simplifié - va directement à get_file_url
//! Pour comparaison avec Python via fake server
//!
//! Ce test est conçu pour être utilisé avec le fake_qobuz_server.py
//! Les credentials sont hardcodés car le fake server les accepte tous

use anyhow::Result;
use pmoqobuz::api::{QobuzApi, Spoofer};

#[tokio::main]
async fn main() -> Result<()> {
    // Initialiser le logging
    tracing_subscriber::fmt::init();

    println!("=== Test get_file_url (Rust) ===\n");

    // Track ID connu (récupéré du test Python)
    let track_id = "19557883";
    let format_id = 27;

    println!("1. Creating QobuzApi (auto-initializes Spoofer)...");
    let spoofer = Spoofer::new().await?;
    let app_id = spoofer.get_app_id()?;
    let app_secret = spoofer.get_app_secret()?;

    let mut api = QobuzApi::with_raw_secret(&app_id, &app_secret)?;

    // Configurer format_id = 27 (comme Python)
    use pmoqobuz::AudioFormat;
    api.set_format(AudioFormat::Flac_HiRes_192);

    println!("   ✓ API created");
    println!("   App ID: {}", app_id);
    println!("   Format: 27 (Flac_HiRes_192) - same as Python");

    println!("\n2. Logging in...");
    println!("   Using fake credentials (fake server accepts all)");
    let username = "eric@coissac.eu";
    let password = "fake_password";

    let user = api.login(username, password).await?;
    println!("   ✓ Login successful - User ID: {}", user.user_id);
    println!("   Token: {}...", &user.token[..20.min(user.token.len())]);

    println!("\n3. Calling track/getFileUrl...");
    println!("   Track ID: {}", track_id);
    println!("   Format ID: {} (default)", format_id);
    println!("   ⚠️  THIS CALL IS SIGNED - Watch fake server logs!\n");

    match api.get_file_url(track_id).await {
        Ok(stream_info) => {
            println!("   ✓ Success!");
            println!(
                "   URL: {}...",
                &stream_info.url[..80.min(stream_info.url.len())]
            );
            println!("   MIME type: {}", stream_info.mime_type);
        }
        Err(e) => {
            println!("   ✗ Failed: {}", e);
            return Err(e.into());
        }
    }

    println!("\n=== Test completed successfully! ===");
    Ok(())
}
